pytest_mpl/plugin.py

```diff
@@ -521,32 +618,57 @@ def compare_image_to_hash_library(self, item, fig, result_dir, summary=None):
         hash_library_filename = (Path(item.fspath).parent / hash_library_filename).absolute()
 
         if not Path(hash_library_filename).exists():
-            pytest.fail(f"Can't find hash library at path {hash_library_filename}")
+            pytest.fail(f"Can't find hash library at path {str(hash_library_filename)!r}.")
 
         hash_library = self.load_hash_library(hash_library_filename)
+        kernel_name = hash_library.get(META_HASH_LIBRARY_KERNEL)
+        if kernel_name is None:
+            msg = (f'Hash library {str(hash_library_filename)!r} missing a '
+                   f'{META_HASH_LIBRARY_KERNEL!r} entry. Assuming that a '
+                   f'{self.kernel.name!r} kernel generated the library.')
+            self.logger.info(msg)
+        else:
+            if kernel_name not in kernel_factory:
+                emsg = (f'Unrecognised hashing kernel {kernel_name!r} specified '
+                        f'in the hash library {str(hash_library_filename)!r}.')
+                pytest.fail(emsg)
+            if kernel_name != self.kernel.name:
+                # TODO: we could be lenient here by raising a warning and hot-swap to
+                #       use the hash library kernel, instead of forcing a test failure?
+                emsg = (f'Hash library {str(hash_library_filename)!r} kernel '
+                        f'{kernel_name!r} does not match configured runtime '
+                        f'kernel {self.kernel.name!r}.')
+                pytest.fail(emsg)
+
```

Comment:
It's important to have the name of the `kernel` that created the hash library as metadata within the hash library.

This will avoid situations where the configured runtime `kernel` is different to the `kernel` that generated the hashes in the library.

Note that, the choice of `META_HASH_LIBRARY_KERNEL` i.e., `pytest-mpl-kernel`, intentionally can't parse as a Python variable, and thus means that it's safe to inject this key into the dictionary, as it's simply not possible for a user of `pytest-mpl` to decorate a test with such a name... well, that's my thinking anyways :wink:

Reply:
Should we also record the high frequency factor used to generate the hash library? (We can probably determine the hash size used from the hashes themselves.) That way, similar to what I was saying above about the kernel name, we can dynamically set the default hash size and high frequency factor to the values used when generating the hash library (only if these have not been explicitly configured by the user as the CLI or ini option). 

What do you think of this approach of getting the default hash settings from the hash library?

Reply:
Done. See https://github.com/matplotlib/pytest-mpl/pull/150/commits/5958c5157a2b583c412132a3568510d1ccee2e6b :+1: